File: server/ExpenseTracker/ExpenseTracker.Application/Dtos/Account/AccountResponseDto.cs
using ExpenseTracker.Application.Dtos.Money;
using ExpenseTracker.Domain.Entities;

namespace ExpenseTracker.Application.Dtos.Account;
public class AccountResponseDto
{
    public int Id { get; set; }
    public string OwnerId { get; set; } = default!;
    public string Name { get; set; } = default!;
    public AccountType Type { get; set; } = default!;
    public MoneyResponseDto Balance { get; set; } = default!;
}

File: clients/web/expense-tracker/src/domain/models/account/AccountResponseDto.ts
import MoneyResponseDto from "../money/MoneyResponseDto";
import AccountType from "./AccountType";

export default class AccountResponseDto {
  Id!: number;

  OwnerId!: string;

  Name!: string;

  Type!: AccountType;

  Balance!: MoneyResponseDto;

  public constructor(init?: Partial<AccountResponseDto>) {
    Object.assign(this, init);
  }
}
